examples/react/nextjs-suspense-streaming/src/app/page.tsx

```diff
@@ -0,0 +1,89 @@
+'use client'
+import { useQuery } from '@tanstack/react-query'
+import { Suspense } from 'react'
+
+// export const runtime = "edge"; // 'nodejs' (default) | 'edge'
+
+function getBaseURL() {
+  if (typeof window !== 'undefined') {
+    return ''
+  }
+  if (process.env.VERCEL_URL) {
+    return `https://${process.env.VERCEL_URL}`
+  }
+  return 'http://localhost:3000'
+}
+const baseUrl = getBaseURL()
+function useWaitQuery(props: { wait: number }) {
+  const query = useQuery({
+    queryKey: ['wait', props.wait],
+    queryFn: async () => {
+      const path = `/api/wait?wait=${props.wait}`
+      const url = baseUrl + path
+
+      console.log('fetching', url)
+      const res: string = await (
+        await fetch(url, {
+          cache: 'no-store',
```

Comment:
Do we need this `no-store`-directive even if it's a client component?

Reply:
I dont think it does anything on the client